execution/src/execution.cc

```diff
@@ -261,15 +277,30 @@ namespace leatherman { namespace execution {
 
         out_stream.open(out_file.c_str(), std::ios::binary);
         if (!out_stream.is_open()) {
-            throw execution_exception(_("failed to open the output file."));
+            throw execution_exception(_("failed to open output file {1}", out_file));
+        }
+
+        boost::system::error_code ec;
+        if (perms) {
+            fs::permissions(out_file, *perms, ec);
+            if (ec) {
+                throw execution_exception(_("failed to modify permissions on output file {1} to {2,num,oct}: {3}", out_file, *perms, ec.message()));
+            }
         }
```

Comment:
@mruzicka going to leave that up to you if you want to add it. The behavior of creating the file then modifying permissions matches Puppet's, but it would be an improvement to set on creation. And we could use the `open` function directly for Windows.